### PR TITLE
Use secretSync

### LINKED ISSUE #99: Use tokens.secretSync instead of .secret
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Use .secretSync() instead of .secret() to generate the secret.

### Motivation

secretSync is 4-5 times faster than .secret(). secret and secretSync are basically wrappers for crypto.randomBytes(secretLength).toString('base64url'). And when we compare to calls of randomBytes throughout the fastify ecosystem, then we can see that we use the "sync" call of randomBytes

see:
https://github.com/fastify/fastify-helmet/blob/b341c6879897c25f6fe3dbd0c7ae239760ed70d0/index.js#L83

So I assume we should call secretSync for better performance. 

### Example

_No response_

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #99

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ async function csrfPlugin (fastify, opts) {
       : this.request.cookies[cookieKey]
     const userInfo = opts ? opts.userInfo : undefined
     if (!secret) {
-      secret = await tokens.secret()
+      secret = await tokens.secretSync()
```
Comment (kibertoad) [inline index.js:62]: should it still be awaited?

Comment (Uzlopak) [reply index.js]: i also removed the async, i hope thats correct?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Note that there is a cache for random data for `randomInt` but nothing for `randomBytes` in core. You might want to open a PR there to introduce that too

**Pull request merged**